### PR TITLE
Feature/13 azure pipelines

### PR DESCRIPTION
**Pull request by alex-flores-jalafund**
Feature/13 azure pipelines

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: Let's remove all above this one

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+trigger:
+  batch: true
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: No need of batch tasks this time, delete please

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  displayName: 'Install DotNet'
+
+- task: DotNetCoreCLI@2
+  inputs:
+    command: 'build'
+    projects: '**/**/*.csproj'
+    arguments: '--configuration $(buildConfiguration)'
+  displayName: 'DotNet Build'
+  
+
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: Blank line

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+- task: PublishBuildArtifacts@1
+  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+  inputs:
+    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
+    ArtifactName: 'drop'
+    publishLocation: 'Container'
+  displayName: 'DotNet Drop'
+
+
+
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: Blank lines

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+- task: DotNetCoreCLI@2
+  inputs:
+    command: 'publish'
+    publishWebProjects: false
+    projects: '**/**/*.csproj'    
+    arguments: '-o $(Build.ArtifactStagingDirectory)/Output'
+    zipAfterPublish: False
+  displayName: 'DotNet Publish'
+
+- task: DotNetCoreCLI@2
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: Remove the pack task please

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+# https://aka.ms/yaml
+
+trigger:
+  batch: true
+  branches:
+    include:
+      - main
+      - develop
+
+pr:
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:7]: please see webapp pipeline to avoid running for draft prs

File: azure-pipelines.yml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      - develop
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+- task: UseDotNet@2
+  inputs:
+    packageType: 'sdk'
+    version: '5.0.x'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: 5.x please

File: azure-pipelines.yml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  displayName: 'Install DotNet'
+
+- task: DotNetCoreCLI@2
+  inputs:
+    command: 'build'
+    projects: '**/**/*.csproj'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: Do we need to dig so deep to get the file? Mmmm

File: azure-pipelines.yml [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    arguments: '--configuration $(buildConfiguration)'
+  displayName: 'DotNet Build'
+  
+
+- task: DotNetCoreCLI@2
+  inputs:
+    command: 'publish'
+    publishWebProjects: false
+    projects: '**/**/*.csproj'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: You need to take this value out to a variable. Furthermore, is there a way we can use the .sln file instead?

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]
Good job! Please check why dotnet 5 is not being picked up in the pipeline tasks :(

File: azure-pipelines.yml (hunk 2 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: ok 
it's done

File: azure-pipelines.yml (hunk 3 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: it's done

File: azure-pipelines.yml (hunk 4 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: it's done
I have deleted that line blank

File: azure-pipelines.yml (hunk 5 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: ok
it's done

File: azure-pipelines.yml (hunk 6 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml:7]: it's done, I added draft option

File: azure-pipelines.yml (hunk 7 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: it's done
3.x version

File: azure-pipelines.yml (hunk 8 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: I have resolved, I changed project to a variable with ../**.sln project

File: azure-pipelines.yml (hunk 9 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: I have resolved, I changed project to a variable with ../**.sln project

File: azure-pipelines.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  branches:
+    include:
+      - develop
+      - main
+
+variables:
+  projects: '**/**/*.sln'
+
+pr:
+  autoCancel: true
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: remove autocancel directive

File: azure-pipelines.yml [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+- task: UseDotNet@2
+  inputs:
+    packageType: 'sdk'
+    version: '3.x'
+  displayName: 'Install DotNet'
+  condition:
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: remove condition

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]
please address obs

File: azure-pipelines.yml (hunk 10 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: It's done

File: azure-pipelines.yml (hunk 11 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: ok 
It's done

File: azure-pipelines.yml [hunk 12]
```diff
@@ -0,0 +1,34 @@
+trigger:
+  branches:
+    include:
+      - develop
+      - main
+variables:
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: add a space that separates triggers and variables
**and please do the same everywhere necessary**
you can check Projects pipeline as referenec

File: azure-pipelines.yml [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    projects: $(projects)
+    arguments: '--output $(Build.ArtifactStagingDirectory)'
+  displayName: 'DotNet Build'
+- task: PublishBuildArtifacts@1
+  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+  inputs:
+    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
+    ArtifactName: 'Drop'
+    publishLocation: 'Container'
+  displayName: 'DotNet Drop'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: add an empty EOL for me, please

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml (hunk 12 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: it's done

File: azure-pipelines.yml (hunk 13 above)
Comment (alex-flores-jalafund) [reply azure-pipelines.yml]: ok
it's done

**Review by martin-callejas-jalafund — APPROVED** [review]
LGTM

**Pull request merged**